acasclient/acasclient.py

```diff
@@ -960,7 +960,90 @@ def get_protocols_by_label(self, label):
                                 .format(self.url, label))
         resp.raise_for_status()
         return resp.json()
+    
+    def get_protocol_by_code(self, protocol_code):
+        """Get a protocol from a protocol code
+
+        Get a protocol given a protocol code
+
+        Args:
+            protocol_code (str): A protocol code
+
+        Returns: Returns a protocol object
+        """
+        resp = self.session.get("{}/api/protocols/codename/{}"
+                                .format(self.url, protocol_code))
+        if resp.status_code == 500:
+            return None
+        resp.raise_for_status()
+        return resp.json()
+
+    def get_all_protocols(self):
+        """Get all protocols
 
+        Get an array of all protocols
+
+        Args:
+            None
+
+        Returns: Returns an array of protocols
+        """
+        resp = self.session.get("{}/api/protocolCodes"
```

Comment:
Does this API really return protocols, not protocol codes?
If I'm reading the code of `exports.protocolCodeList` correctly, it returns little codeTable objects of code, name, and ignored.

I find this acasclient endpoint to be confusingly named - I'd expect it to return full protocol objects, with the same structure that `get_protocol_by_code` returns.

Reply:
Thanks for pointing that out.  This was AI copy pasta that I did not catch. I just added the real return info in the doc string.

Reply:
That's helpful. I still do find the function name `get_all_protocols` misleading though. Maybe `get_all_protocol_codetables` or `get_all_protocol_stubs`?
Those dicts are not `Protocol`s are they?

Reply:
Just updated it to be get_all_protocol_stubs